androidapp/Connexus/app/src/main/java/com/green/apt/connexus/ViewSearchActivity.java

```diff
@@ -0,0 +1,181 @@
+import android.widget.EditText;
+import android.widget.GridView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONArray;
+
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ViewSearchActivity extends AppCompatActivity {
+
+    private static final String BASE_URL = "https://apt-miniproject-greenteam-v2.appspot.com/api/";
+    private static String relativeUrl = "search";
+    private static String streams = "Search Results";
+    private static String searchQuery = "";
+    private static int offset = 0;
+    private static List<String> streamCoverUrls = new ArrayList<>();
+    private static List<Long> streamIds = new ArrayList<>();
+    private static List<String> streamNames = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_view_search);
+        Intent intent = getIntent();
+        Bundle extras = intent.getExtras();
+        String query = intent.getStringExtra("SEARCH_QUERY");
+
+        runSearch(query);
+
+    }
+
+    private void updateTextFields(String query){
+        // Capture the layout's TextView and set the string as its text
+        TextView searchText = (TextView) findViewById(R.id.searchText);
+        searchText.setText(query);
+
+        EditText searchField = (EditText) findViewById(R.id.searchField);
+        searchField.setText(query);
+    }
+    private void updateResultCount(){
+        TextView searchText = (TextView) findViewById(R.id.resultCt);
+        int resultSize = streamCoverUrls.size();
+        searchText.setText(String.valueOf(resultSize));
+    }
+
+    public void searchAgain(View view){
+        this.hideKeyboard();
+        EditText searchField = (EditText) findViewById(R.id.searchField);
+        String query = searchField.getText().toString();
+        offset = 0;
+        runSearch(query);
+    }
+
+    public void viewNext(View view){
+        offset += 8;
+        TextView searchText = (TextView) findViewById(R.id.searchText);
+        String query = searchText.getText().toString();
+        runSearch(query);
+    }
+
+    private void runSearch(String query){
+        try{
+            searchQuery = URLEncoder.encode(query, "UTF-8");
+        }
+        catch(Exception e) {
+            Log.d("ViewSearchActivity", "Errored encoding: " + query);
+            Log.e("ViewSearchActivity", e.getMessage());
+        }
+        streamCoverUrls.clear();
+        streamIds.clear();
+        streamNames.clear();
+
+        this.updateTextFields(query);
+
+        this.getSearchStreams();
+    }
+
+    private void hideKeyboard(){
```

Comment:
Another candidate for the BaseController

Reply:
ok